Beta_Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -0,0 +1,1648 @@
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': contents,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown('Sentinel One Activities', contents, headers, removeNull=True),
+        'EntryContext': context
+    })
+
+
+def get_groups_request(group_type=None, group_ids=None, group_id=None, is_default=None, name=None, query=None,
+                       rank=None, limit=None):
+
+    endpoint_url = 'groups'
+
+    params = {
+        'type': group_type,
+        'groupIds': group_ids,
+        'id': group_id,
+        'isDefault': is_default,
+        'name': name,
+        'query': query,
+        'rank': rank,
+        'limit': limit
+    }
+
+    response = http_request('GET', endpoint_url, params)
+    if response.get('errors'):
+        return_error(response.get('errors'))
+    if 'data' in response:
+        return response.get('data')
+    return {}
+
+
+def get_groups_command():
+    """
+    Gets the group data.
+    """
+
+    context = {}
+    contents = []
+    headers = ['ID', 'Name', 'Type', 'Creator', 'Creator ID', 'Created at', 'Rank']
+
+    group_type = demisto.args().get('type')
+    group_id = demisto.args().get('id')
+    group_ids = argToList(demisto.args().get('groupIds', []))
+    is_default = demisto.args().get('isDefault')
+    name = demisto.args().get('name')
+    query = demisto.args().get('query')
+    rank = demisto.args().get('rank')
+    limit = int(demisto.args().get('limit', 50))
+
+    groups = get_groups_request(group_type, group_id, group_ids, is_default, name, query, rank, limit)
+    if groups:
+        for group in groups:
+            contents.append({
+                'ID': group.get('id'),
+                'Type': group.get('type'),
+                'Name': group.get('name'),
+                'Creator ID': group.get('creatorId'),
+                'Creator': group.get('creator'),
+                'Created at': group.get('createdAt'),
+                'Rank': group.get('rank')
+            })
+
+        context['SentinelOne.Group(val.ID && val.ID === obj.ID)'] = groups
+
+    demisto.results({
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': contents,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown('Sentinel One Groups', contents, headers, removeNull=True),
+        'EntryContext': context
+    })
+
+
+def delete_group_request(group_id=None):
+
+    endpoint_url = 'groups/' + group_id
+
+    response = http_request('DELETE', endpoint_url)
+    if response.get('errors'):
+        return_error(response.get('errors'))
+    if 'data' in response:
+        return response.get('data')
+    return {}
+
+
+def delete_group():
+    """
+    Deletes a group by ID.
+    """
+    group_id = demisto.args().get('group_id')
+
+    delete_group_request(group_id)
+    demisto.results('The group was deleted successfully')
```

Comment:
Are we sure there is no situation where you don't get error in response yet the call still fails?
If our only indication for success is 200 response then you can resolve.